controllers/api/v3.js

```diff
@@ -2136,6 +2136,63 @@ apiController.search = async function(request, response) {
     });
 };
 
+/**
+ * Responds with an array of MAC addresses of vendor mesh master devices that
+ * are inside requested page with requested page size, limited to MAX_PAGE_SIZE.
+ *
+ * @memberof controllers/api/v3
+ *
+ * @async
+ *
+ * @param {HTTPRequest} req - The HTTP request.
+ * @param {HTTPResponse} res  - The HTTP response.
+ */
+apiController.getVendorMeshDevices = async function(req, res) {
+  let success = false; // default success state for this request.
+
+  // Reading request parameters.
+  const page = parseInt(req.params.page);
+  const pageLimit = req.params.pageLimit ?
+    Math.min(parseInt(req.params.pageLimit), MAX_PAGE_SIZE) : MAX_PAGE_SIZE;
+  if ( // Checking request parameters.
+    isNaN(page) || isNaN(pageLimit) || // If parameters are not numeric.
+    page < 1 || pageLimit < 1 // If parameters are zero or less.
+  ) {
+    return res.status(400).json({
+      success, message: t('requestError', {errorline: __line}),
+    });
+  }
+
+  const query = {mesh_vendor: 10}; // Definition of vendor mesh master device.
+
+  const ret = await Promise.all([ // Executing queries concurrently.
+    // Querying for vendor mesh masters and returning only the ids.
+    DeviceModel.find(query, {_id: 1})
+      .skip((page-1)*pageLimit).limit(pageLimit).lean().exec(),
+    // Getting amount of documents that could be returned.
+    DeviceModel.countDocuments(query),
+  ]).catch((e) => e);
```

Comment:
`DeviceModel.paginate` já faz isso tudo para você

Reply:
E faz outras coisas que não são necessária. Esse código é mais limpo do que o pacote importado.

Reply:
Todo o código do Flashman que precisa de paginação usa a chamada de `paginate`. Só porque o pacote faz um monte de coisa, não quer dizer que seja menos eficiente ou que não deveria ser usado. Discordo fortemente sobre o código ser mais limpo - só tem menos linhas, a alternativa deixa muito mais claro o que está sendo feito sem ter que parsear ou dar manutenção em lógica:
```suggestion
  const paginateOptions = {
    page: page,
    limit: pageLimit,
    projection: {_id: 1},
    lean: true,
  };
  const ret = await Promise.all([ // Executing queries concurrently.
    // Querying for vendor mesh masters and returning only the ids.
    DeviceModel.paginate(query, paginateOptions).exec(),
    // Getting amount of documents that could be returned.
    DeviceModel.countDocuments(query),
  ]).catch((e) => e);
```

Reply:
Coloquei o paginate.